src/app/reactUtils.ts

```diff
@@ -12,3 +13,26 @@ export const useDrawFocus = <E extends HTMLElement = HTMLElement>() => {
 
   return ref;
 };
+
+export const onFocusLostHandler = (ref: React.RefObject<HTMLElement>, isEnabled: boolean, cb: () => void) => () => {
+  const el = ref && ref.current;
+  if (!el) { return; }
+
+  const handler = (ev: FocusEvent) => {
+    const relatedTarget = ev.relatedTarget as HTMLElement | null;
+
+    if (!relatedTarget || !elementDescendantOf(relatedTarget, ref.current!)) {
+      cb();
+    }
+  };
+
+  if (isEnabled) {
+    el.addEventListener('focusout', handler as any);
+  }
+
+  return () => el.removeEventListener('focusout', handler as any);
+};
+
+export const useFocusLost = (ref: React.RefObject<HTMLElement>, isEnabled: boolean, cb: () => void) => {
+  React.useEffect(onFocusLostHandler(ref, isEnabled, cb), [ref, isEnabled, cb]);
```

Comment:
i think you want this to be like this
```suggestion
  React.useEffect(onFocusLostHandler(ref, isEnabled, cb), [isEnabled]);
```

Reply:
So, there is a lint error about `react-hooks/exhaustive-deps` which always make me wonder if it is useful. From one perspective, if we know that those args will not change then there is no reason to add them to dependencies array, but from the other perspective they can be dynamically set in some other case in the future and then keeping them inside will be helpful.
Eventually, I've ended up adding those deps depending on the case. Here we have `ref` which actually can change so hook should be called one more time after this happens and we have `cb` which most likely will not change, but in this case, adding it to dependency array will not have any side effects so I guess it's okay. 
Things get complicated when you are using like 6 dependencies inside hook but you know they will not change. In this case I will most likely add only those which are needed for sure.

If you would like to have other guidelines for this lint error just inform me

Reply:
the callback will be different every time the hook is called. agree that the ref is probably safe to keep

Reply:
Ahh, right, because we are creating new function instead of passing a reference. I've removed `cb` from dependencies array